File: agents/GuiConversion/Wx01.cpp
#ifdef HAVE_WX
#include <wx/wx.h>
#include <wx/app.h>

class HelloWxApp : public wxApp {
public:
    bool OnInit() override { return true; }
};

wxIMPLEMENT_APP_NO_MAIN(HelloWxApp);

void Wx01(int test_num) {
    int argc = 1;
    char arg0[] = "wxapp";
    char* argv[] = { arg0 };
    
    if (test_num == 0 || test_num < 0) {
        if (!wxEntryStart(argc, argv))
            return;
        if (!wxTheApp || !wxTheApp->CallOnInit()) { wxEntryCleanup(); return; }

        wxFrame* frame = new wxFrame(nullptr, wxID_ANY, "Hello World program",
                                      wxDefaultPosition, wxSize(320, 240));
        wxPanel* panel = new wxPanel(frame);
        wxBoxSizer* sizer = new wxBoxSizer(wxVERTICAL);
        auto* label = new wxStaticText(panel, wxID_ANY, "Hello world!");
        sizer->AddStretchSpacer(1);
        sizer->Add(label, 0, wxALIGN_CENTER | wxALL, 0);
        sizer->AddStretchSpacer(1);
        panel->SetSizerAndFit(sizer);
        frame->Centre();
        frame->Show(true);
        wxTheApp->SetTopWindow(frame);

        wxTheApp->OnRun();

        frame->Destroy();
        wxTheApp->OnExit();
        wxEntryCleanup();
    }

    // 2. Simple events (Button & click -> popup)
    if (test_num == 1 || test_num < 0) {
        if (!wxEntryStart(argc, argv))
            return;
        if (!wxTheApp || !wxTheApp->CallOnInit()) { wxEntryCleanup(); return; }

        wxFrame* frame = new wxFrame(nullptr, wxID_ANY, "Button program",
                                      wxDefaultPosition, wxSize(320, 240));
        wxPanel* panel = new wxPanel(frame);
        wxButton* button = new wxButton(panel, wxID_HIGHEST + 1, "Hello world!",
                                        wxPoint(30, 30), wxSize(100, 30));
        panel->Bind(wxEVT_BUTTON, [frame](wxCommandEvent&) {
            wxMessageBox("Popup message", "", wxOK | wxICON_INFORMATION, frame);
        }, button->GetId());

        frame->Centre();
        frame->Show(true);
        wxTheApp->SetTopWindow(frame);

        wxTheApp->OnRun();

        frame->Destroy();
        wxTheApp->OnExit();
        wxEntryCleanup();
    }
}
#else
void Wx01(int) {}
#endif

File: agents/LangGuiConversion/Wx01.py

# 1. Hello World GUI (wxPython)
# 2. Simple button with click -> message box
import wx


def hello_world():
    app = wx.App(False)
    frame = wx.Frame(None, title="Hello World program", size=(320, 240))
    panel = wx.Panel(frame)
    sizer = wx.BoxSizer(wx.VERTICAL)
    label = wx.StaticText(panel, label="Hello world!")
    sizer.AddStretchSpacer(1)
    sizer.Add(label, 0, wx.ALIGN_CENTER)
    sizer.AddStretchSpacer(1)
    panel.SetSizer(sizer)
    frame.Centre()
    frame.Show()
    app.MainLoop()


def button_demo():
    app = wx.App(False)
    frame = wx.Frame(None, title="Button program", size=(320, 240))
    panel = wx.Panel(frame)
    btn = wx.Button(panel, label="Hello world!", pos=(30, 30), size=(100, 30))

    def on_click(_evt):
        wx.MessageBox("Popup message", "", wx.OK | wx.ICON_INFORMATION, parent=frame)

    btn.Bind(wx.EVT_BUTTON, on_click)
    frame.Centre()
    frame.Show()
    app.MainLoop()
